### ISSUE TITLE
Docker Installation error 

### ISSUE THREAD

**Issue by Vaibhav-R-13** [docker/for-win#12511]
Docker Version: 4.4.2 (73305)
Sha1: 
Started on: 2022/01/24 10:18:43.858
Resources: C:\Users\vr\Downloads\resources
OS: Windows 10 Enterprise
Edition: Enterprise
Id: 2009
Build: 19042
Getting below error while installing docker desktop for windows 10.
```
Component CommunityInstaller.ExecAction failed: The requested operation requires elevation
   at CommunityInstaller.InstallWorkflow.<DoHandleD4WPackageAsync>d__30.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at CommunityInstaller.InstallWorkflow.<DoProcessAsync>d__24.MoveNext()
```
Also attaching log for it
[install-log.txt](https://github.com/docker/for-win/files/7923212/install-log.txt)

**Comment by mat007** [comment]
The installer should pop up a UAC confirmation box, I’m not sure why it doesn’t do that for you…
Does it work if you run in as admin (right mouse button, «Run as administrator»)?

**Comment by Vaibhav-R-13** [comment]
@mat007 I run it as admin only and didn't get any UAC confirmation box.

**Comment by mat007** [comment]
It’s when running it as non-admin that there is a UAC prompt.
Are you saying the error you pasted in the issue description above is from when you run the installer as admin?

**Comment by mat007** [comment]
Also are you updating an existing Docker Desktop (which version?) or is this a fresh installation?
Do you have any files in `C:\Program Files\Docker\Docker`?

**Comment by Vaibhav-R-13** [comment]
I tried to install it both ways admin and without it. I am installing it fresh no previous version is there also once the installation fails it removes all the files so the C:\Program Files\Docker is empty. I also tried to remove the stuff from C:\Users\vr\AppData\Local\Docker and C:\Users\vr\AppData\Roaming\Docker but no luck.

**Comment by mat007** [comment]
Do you have no other log file in `%LOCALAPPDATA%\Docker`?

**Comment by mat007** [comment]
@aniskom is that the only log file in `%LOCALAPPDATA%\Docker`?

**Comment by aniskom** [comment]
@mat007 yes. There is only one log file.

**Comment by mat007** [comment]
@aniskom are you seeing a UAC prompt request admin elevation when running the installer?
If not, I suspect your system may have been configured to application requests for elevation, https://docs.microsoft.com/en-us/windows/security/identity-protection/user-account-control/user-account-control-group-policy-and-registry-key-settings lists the various possible settings.
One think you both have in common is your running an Enterprise version of Windows.

Clearly the Docker Desktop installer should detect that elevation fails and display a proper message.
What I don’t understand is how you can still get the same error if you manually force the installer to «run as administrator» 🤔

**Comment by aniskom** [comment]
@mat007 actually it all worked fine when installing through winget. 

Unfortunately nothing helped when I tried to run the installer the normal way.

**Comment by mat007** [comment]
@aniskom that’s interesting!

So, running the installer directly you never saw a UAC popup asking for admin rights?

**Comment by aniskom** [comment]
@mat007 no. I ran the installer with administrator rights and nothing popped up.

**Comment by mat007** [comment]
The UAC popup only appears if you run the installer *without* administrator rights.

**Comment by Vaibhav-R-13** [comment]
@mat007 No UAC pop up for me as well
Thanks @aniskom  I am also able to install it through winget.

**Comment by jazzyray** [comment]
The winget output -->

Installing components 0%
[12:23:10.402][InstallWorkflow   ][Info   ] Deploying component CommunityInstaller.CreateGroupAction
Deploying component: Create docker-users group 0%
/[12:23:24.478][InstallWorkflow   ][Info   ] Deploying component CommunityInstaller.AddToGroupAction
Deploying component: Add user to docker-users group 0%
[12:23:24.486][InstallWorkflow   ][Info   ] Deploying component CommunityInstaller.EnableFeaturesAction
Deploying component: Enable Hyper-V Windows Features 0%
[12:23:24.513][InstallWorkflow   ][Error  ] Installation failed
Exception type: System.Exception, Exception message: Component CommunityInstaller.EnableFeaturesAction failed: Not found , StackTrace:
   at CommunityInstaller.InstallWorkflow.<DoHandleD4WPackageAsync>d__30.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at CommunityInstaller.InstallWorkflow.<DoProcessAsync>d__24.MoveNext()
[12:23:24.516][InstallWorkflow   ][Info   ] Rollbacking component CommunityInstaller.AddToGroupAction
Rollbacking component: Add user to docker-users group 0%
[12:23:24.518][InstallWorkflow   ][Info   ] Rollbacking component CommunityInstaller.CreateGroupAction
Rollbacking component: Create docker-users group 0%
[12:23:24.519][InstallWorkflow   ][Info   ] Rollbacking component CommunityInstaller.UnpackArtifactsStep
Rollbacking component:  0%
Removing files 0%
[12:23:24.527][FileSystem        ][Info   ] Deleting C:\Program Files\Docker\Docker
|[12:23:25.229][FileSystem        ][Info   ] Deleted C:\Program Files\Docker\Docker successfully
[12:23:25.229][InstallWorkflow   ][Info   ] Rollbacking component CommunityInstaller.ExtractManifestStep
Rollbacking component:  0%
[12:23:25.230][InstallWorkflow   ][Info   ] Rollbacking component CommunityInstaller.DownloadStep
Installation failed
Component CommunityInstaller.EnableFeaturesAction failed: Not found
                                                                       at CommunityInstaller.InstallWorkflow.<DoHandleD4WPackageAsync>d__30.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at CommunityInstaller.InstallWorkflow.<DoProcessAsync>d__24.MoveNext()
Installer failed with exit code: 1

**Comment by jon-laudner** [comment]
I am having the same issue.

Clean install, tried installing with Admin & Non-Admin, neither gave me the UAT pop up, also tried via chocolatey in PowerShell and received the exact same error that way too.  I was in PowerShell as Admin.

2022-03-03 11:09:52,506 18384 [ERROR] - [10:39:53.070][InstallWorkflow   ][Info   ] Deploying component CommunityInstaller.ExecAction
2022-03-03 11:09:52,508 18384 [ERROR] - Exception type: System.Exception, Exception message: Component CommunityInstaller.ExecAction failed: The requested operation requires elevation, StackTrace:
2022-03-03 11:09:52,511 18384 [ERROR] - [10:39:53.293][InstallWorkflow   ][Error  ] Installation failed
2022-03-03 11:09:52,513 18384 [ERROR] -    at CommunityInstaller.InstallWorkflow.<DoHandleD4WPackageAsync>d__30.MoveNext()
2022-03-03 11:09:52,515 18384 [ERROR] - --- End of stack trace from previous location where exception was thrown ---
2022-03-03 11:09:52,516 18384 [ERROR] -    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
2022-03-03 11:09:52,519 18384 [ERROR] -    at CommunityInstaller.InstallWorkflow.<DoProcessAsync>d__24.MoveNext()
2022-03-03 11:09:52,522 18384 [ERROR] -    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)

**Comment by jazzyray** [comment]
Hey docker, I have raised a support ticket....and I am not getting any help...I am bounced back and forth...and no solutions or help to diagnose...this ticket has also been open for a while with no solutions or ideas on how to diagnose?

The log output in your installer is not providing enough information to be able to understand what the root cause and problem is.

This does is not very helpful?

2022-03-03 11:09:52,508 18384 [ERROR] - Exception type: System.Exception, Exception message: Component CommunityInstaller.ExecAction failed: The requested operation requires elevation, StackTrace:
2022-03-03 11:09:52,511 18384 [ERROR] - [10:39:53.293][InstallWorkflow ][Error ] Installation failed
2022-03-03 11:09:52,513 18384 [ERROR] - at CommunityInstaller.InstallWorkflow.d__30.MoveNext()
2022-03-03 11:09:52,515 18384 [ERROR] - --- End of stack trace from previous location where exception was thrown ---
2022-03-03 11:09:52,516 18384 [ERROR] - at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
2022-03-03 11:09:52,519 18384 [ERROR] - at CommunityInstaller.InstallWorkflow.d__24.MoveNext()
2022-03-03 11:09:52,522 18384 [ERROR] - at


Will you be providing a release with better diagnosis logs so this can be resolved?
Cheers
Jem

**Comment by mat007** [comment]
Hi @jazzyray,

Your error is obviously different from the one experienced by the others in the issue (I think in your last comment you just copy-pasted from the previous comment, didn’t you?).

That being said, can you open powershell and type `Import-Module Dism;` then enter.
Does it give you an error?

**Comment by jazzyray** [comment]
@mat007 my error certainly appears to be very similar to @jon-laudner and @aniskom ?

You will notice

```
at CommunityInstaller.InstallWorkflow.d__30.MoveNext()
```

is included in all three logs, without any further detail. Am missing something? 

Regardless the errors in all cases do not indicate or provide any detail as to what the underlying error/issue is. The log output is opaque. 

On  "That being said, can you open powershell and type Import-Module Dism; then enter."

This is the output --->

```
PS C:\WINDOWS\system32> Import-Module Dism;
PS C:\WINDOWS\system32>
```

No error was reported.

Cheers
Jem

**Comment by mat007** [comment]
@jazzyray your message [here](https://github.com/docker/for-win/issues/12511#issuecomment-1059078133) has the exact same log as the previous message (same timestamps and everything).

Your two other logs have this error:
> Exception type: System.Exception, Exception message: Component CommunityInstaller.EnableFeaturesAction failed: Not found , 

whereas the other persons in this thread have:
> Exception type: System.Exception, Exception message: Component CommunityInstaller.ExecAction failed: The requested operation requires elevation, StackTrace:

Anyway, that being said, I’ll add some more logs around these two errors and will post a link to a private build.
Hopefully we can understand what these two errors really mean (I see in the code where they’re raised but I’m not sure why, yet).

**Comment by mat007** [comment]
Here is the link to a private build, if you can try it and post the content of `%LOCALAPPDATA%\Docker\install-log.txt` that will help, and we can iterate from there.

Thanks!

**Comment by jon-laudner** [comment]
I'm including two log files, first one is when I tried installing just by double-clicking the exe in my Downloads folder. 
[install-log.txt](https://github.com/docker/for-win/files/8188428/install-log.txt)

I also tried installing via PowerShell as Admin and the log file looks identical but here is that one as well.
Invoke-Command -ScriptBlock {Start-Process "C:\Users\jlaudner\Downloads\Docker Desktop Installer.exe" -Wait}
[install-log-powershell.txt](https://github.com/docker/for-win/files/8188460/install-log-powershell.txt)

**Comment by mat007** [comment]
@jon-laudner what is your UAC level? You can see that in the settings (e.g. Windows key, then type «uac»).

Here is mine:

![image](https://user-images.githubusercontent.com/2386884/156836032-cc4c32d2-a4ab-4165-aafd-402853303c6b.png)

**Comment by jon-laudner** [comment]
I cannot open that on my machine, IT has it locked down.

**Comment by mat007** [comment]
Are you still admin on your machine? How do you usually install software?

**Comment by jon-laudner** [comment]
I am not a Local Admin on my machine, I have to have IT install any software for me, and they tried it for me yesterday when I first got the error.  But I can elevate my priviledges in Powershell as a bit of a workaround.  And speaking of, I was able to elevate my UAC level temporarily and looks like I have Docker installed successfully now!

C:\Users\jlaudner>docker -v
Docker version 20.10.12, build e91ed57

**Comment by mat007** [comment]
Ah great! It does have something to do with UAC. I’m wondering how IT configured the Local Group Policy on your machine.
Can you try to run in a powershell:
```
gpresult /v
```
I’m not sure if this will give you a lot of info if you’re not running it as admin though…

**Comment by jazzyray** [comment]
Hi @mat007 here is the log file using the private build
 
[install-log.txt](https://github.com/docker/for-win/files/8190726/install-log.txt)

I set UAC as 

![image](https://user-images.githubusercontent.com/25301475/156885072-dcccedae-cb0a-4fc1-bda1-3514d1458791.png)

it still failed.

**Comment by jazzyray** [comment]
I also set UAC to replicate your settings @mat007 however it still failed.
Here is the log file. 

Cheers
Jem

[install-log.txt](https://github.com/docker/for-win/files/8190749/install-log.txt)

**Comment by jazzyray** [comment]
Powershell as admin

PS C:\WINDOWS\system32> gpresult /v
ERROR: Access Denied.

![image](https://user-images.githubusercontent.com/25301475/156885613-27b577bb-da54-4220-a9d4-25f1b715c0eb.png)

**Comment by mat007** [comment]
@jazzyray I’m pretty sure UAC is not the issue for you.

Does
```
dism /online /get-features
```
list you a bunch of Hyper-V features?

**Comment by jazzyray** [comment]
[dism.txt](https://github.com/docker/for-win/files/8190766/dism.txt)

Hi @mat007  here is the output

**Comment by mat007** [comment]
That does look fine.
In your services, is the «Windows Management Instrumentation» service running?

**Comment by jazzyray** [comment]
It is

![image](https://user-images.githubusercontent.com/25301475/156886266-fcb020a8-9d9a-4958-8bbb-8dd9f01c740a.png)

**Comment by mat007** [comment]
Can you run (from an elevated shell) `winmgmt /verifyrepository`?

**Comment by jazzyray** [comment]
Hi @mat007 here is the output

C:\WINDOWS\system32>winmgmt /verifyrepository
WMI repository is consistent

**Comment by mat007** [comment]
Sounds good, what about `Get-CimInstance -ClassName Win32_OptionalFeature`?

**Comment by jazzyray** [comment]
Interestingly the class is unknown

```
PS C:\WINDOWS\system32> Get-CimInstance -ClassName Win32_OptionalFeature
Get-CimInstance : Invalid class
At line:1 char:1
+ Get-CimInstance -ClassName Win32_OptionalFeature
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : MetadataError: (root\cimv2:Win32_OptionalFeature:String) [Get-CimInstance], CimException
    + FullyQualifiedErrorId : HRESULT 0x80041010,Microsoft.Management.Infrastructure.CimCmdlets.GetCimInstanceCommand
```

I assume you are using this to check the pre-requiste features are enabled?

**Comment by jazzyray** [comment]
Missing class -->

```
Get-CimClass -ClassName *Win32*


   NameSpace: ROOT/cimv2

CimClassName                        CimClassMethods      CimClassProperties
------------                        ---------------      ------------------
Win32_ComputerSystemEvent           {}                   {SECURITY_DESCRIPTOR, TIME_CREATED, MachineName}
Win32_ComputerShutdownEvent         {}                   {SECURITY_DESCRIPTOR, TIME_CREATED, MachineName, Type}
Win32_SystemTrace                   {}                   {SECURITY_DESCRIPTOR, TIME_CREATED}
Win32_ProcessTrace                  {}                   {SECURITY_DESCRIPTOR, TIME_CREATED, ParentProcessID, ProcessID...}
Win32_ProcessStartTrace             {}                   {SECURITY_DESCRIPTOR, TIME_CREATED, ParentProcessID, ProcessID...}
Win32_ProcessStopTrace              {}                   {SECURITY_DESCRIPTOR, TIME_CREATED, ParentProcessID, ProcessID...}
Win32_ModuleTrace                   {}                   {SECURITY_DESCRIPTOR, TIME_CREATED}
Win32_ModuleLoadTrace               {}                   {SECURITY_DESCRIPTOR, TIME_CREATED, DefaultBase, FileName...}
Win32_ThreadTrace                   {}                   {SECURITY_DESCRIPTOR, TIME_CREATED, ProcessID, ThreadID}
Win32_ThreadStartTrace              {}                   {SECURITY_DESCRIPTOR, TIME_CREATED, ProcessID, ThreadID...}
Win32_ThreadStopTrace               {}                   {SECURITY_DESCRIPTOR, TIME_CREATED, ProcessID, ThreadID}
__Win32Provider                     {}                   {Name, ClientLoadableCLSID, CLSID, Concurrency...}
Win32_CollectionStatistics          {}                   {Collection, Stats}
Win32_NTLogEvent                    {}                   {Category, CategoryString, ComputerName, Data...}
Win32_OfflineFilesUserConfiguration {}                   {AssignedOfflineFiles, IsConfiguredByWMI, MakeAvailableOfflineButtonRemoved, WorkOfflineButtonRemoved}
Win32_AccountSID                    {}                   {Element, Setting}
Win32_WinSAT                        {}                   {CPUScore, D3DScore, DiskScore, GraphicsScore...}
Win32_CurrentTime                   {}                   {Day, DayOfWeek, Hour, Milliseconds...}
Win32_UTCTime                       {}                   {Day, DayOfWeek, Hour, Milliseconds...}
Win32_LocalTime                     {}                   {Day, DayOfWeek, Hour, Milliseconds...}
Win32_FolderRedirectionHealth       {}                   {HealthStatus, LastSuccessfulSyncTime, LastSyncStatus, LastSyncTime...}
Win32_OfflineFilesAssociatedItems   {}                   {Antecedent, Dependent}
Win32_UserProfile                   {ChangeOwner}        {AppDataRoaming, Contacts, Desktop, Documents...}
Win32_OfflineFilesChangeInfo        {}                   {CreatedOffline, DeletedOffline, Dirty, ModifiedAttributes...}
Win32_RoamingProfileMachineConfi... {}                   {AddAdminGroupToRUPEnabled, AllowCrossForestUserPolicy, BackgroundUploadParams, DeleteProfilesOlderDays...}
Win32_SID                           {}                   {AccountName, BinaryRepresentation, ReferencedDomainName, SID...}
Win32_RoamingProfileUserConfigur... {}                   {DirectoriesToSyncAtLogonLogoff, ExcludedProfileDirs, IsConfiguredByWMI}
Win32_ShadowStorage                 {Create}             {AllocatedSpace, DiffVolume, MaxSpace, UsedSpace...}
Win32_FolderRedirection             {}                   {ContentsMoved, ContentsMovedOnPolicyRemoval, ContentsRenamedInLocalCache, ExclusiveRightsGranted...}
Win32_NamedJobObjectProcess         {}                   {Collection, Member}
Win32_NamedJobObject                {}                   {Caption, CollectionID, Description, BasicUIRestrictions}
Win32_OfflineFilesItem              {}                   {ChangeInfo, ConnectionInfo, DirtyInfo, Encrypted...}
Win32_OfflineFilesBackgroundSync    {}                   {BackgroundSyncWorkOfflineSharesEnabled, BlockOutDurationMin, BlockOutStartTimeHoursMinutes, MaxTimeBetweenSyncs...}
Win32_InstalledWin32Program         {}                   {Language, MsiPackageCode, MsiProductCode, Name...}
Win32_RoamingUserHealthConfigura... {}                   {HealthStatusForTempProfiles, LastProfileDownloadIntervalCautionInHours, LastProfileDownloadIntervalUnhealthyInHours, LastProfileUploadInte...
Win32_UserStateConfigurationCont... {}                   {FolderRedirection, OfflineFiles, RoamingUserProfile}
Win32_OfflineFilesPinInfo           {}                   {Pinned, PinnedForComputer, PinnedForFolderRedirection, PinnedForUser...}
Win32_OfflineFilesFileSysInfo       {}                   {LocalAttributes, LocalChangeTime, LocalCreationTime, LocalLastAccessTime...}
Win32_DfsNode                       {Create}             {Caption, Description, InstallDate, Name...}
Win32_ShadowProvider                {}                   {Caption, Description, InstallDate, Name...}
Win32_ShadowCopy                    {Create, Revert}     {Caption, Description, InstallDate, Name...}
Win32_DfsTarget                     {}                   {Caption, Description, InstallDate, Name...}
Win32_NTEventlogFile                {TakeOwnerShip, C... {Caption, Description, InstallDate, Name...}
Win32_OfflineFilesSuspendInfo       {}                   {Suspended, SuspendedRoot}
Win32_PingStatus                    {}                   {Address, BufferSize, NoFragmentation, PrimaryAddressResolutionStatus...}
Win32_FolderRedirectionUserConfi... {}                   {AppDataRoaming, Contacts, Desktop, Documents...}
Win32_Reliability                   {}                   {}
Win32_ReliabilityStabilityMetrics   {GetRecordCount}     {EndMeasurementDate, RelID, StartMeasurementDate, SystemStabilityIndex...}
Win32_ReliabilityRecords            {GetRecordCount}     {ComputerName, EventIdentifier, InsertionStrings, Logfile...}
Win32_InstalledProgramFramework     {}                   {FrameworkName, FrameworkPublisher, FrameworkVersion, FrameworkVersionActual...}
Win32_NTLogEventLog                 {}                   {Log, Record}
Win32_DiskQuota                     {}                   {DiskSpaceUsed, Limit, QuotaVolume, Status...}
Win32_FolderRedirectionHealthCon... {}                   {LastSyncDurationCautionInHours, LastSyncDurationUnhealthyInHours}
Win32_OfflineFilesMachineConfigu... {}                   {BackgroundSyncEnabled, BackgroundSyncParams, DiskSpaceLimitEnabled, DiskSpaceLimitParams...}
Win32_OfflineFilesCache             {Enable, RenameIt... {Active, Enabled, Location}
Win32_NTLogEventUser                {}                   {Record, User}
Win32_OfflineFilesConnectionInfo    {}                   {ConnectState, OfflineReason}
Win32_InstalledStoreProgram         {}                   {Architecture, Language, Name, ProgramId...}
Win32_NTLogEventComputer            {}                   {Computer, Record}
Win32_RoamingProfileBackgroundUp... {}                   {Interval, SchedulingMethod, Time}
Win32_RoamingProfileSlowLinkParams  {}                   {ConnectionTransferRate, TimeOut}
Win32_OfflineFilesDiskSpaceLimit    {}                   {AutoCacheSizeInMB, TotalSizeInMB}
Win32_OfflineFilesDirtyInfo         {}                   {LocalDirtyByteCount, RemoteDirtyByteCount}
Win32_OfflineFilesHealth            {}                   {LastSuccessfulSyncTime, LastSyncStatus, LastSyncTime, OfflineAccessEnabled...}
```

**Comment by jazzyray** [comment]
fwiw: 

Checking optional features using the Get-WindowsOptionalFeature works OK ->

```
 Get-WindowsOptionalFeature -Online -FeatureName 'Microsoft-Hyper-V-Hypervisor'


FeatureName      : Microsoft-Hyper-V-Hypervisor
DisplayName      : Hyper-V Hypervisor
Description      : Provides the Hyper-V Hypervisor.
RestartRequired  : Possible
State            : Enabled
CustomProperties :
```

**Comment by Orbazz** [comment]
> The installer should pop up a UAC confirmation box, I’m not sure why it doesn’t do that for you… Does it work if you run in as admin (right mouse button, «Run as administra

**Comment by Orbazz** [comment]
am having the same issues right now , please what can i do i have tried all the alternative i know, even youtube. please can you explain more futher

**Comment by woutkonings** [comment]
I am having issues again since an overnight update of Windows. My Docker Desktop failed to startup, so I decided to reinstall. Now the reinstall is failing. I am on Microsoft Windows [Version 10.0.19044.1826] trying to install Docker Desktop 4.10.1 and getting the following error:

`Component CommunityInstaller.EnableFeaturesAction failed: Not found 
   with CommunityInstaller.InstallWorkflow.<DoHandleD4WPackageAsync>d__30.MoveNext()
   with System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   with System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   with CommunityInstaller.InstallWorkflow.<DoProcessAsync>d__23.MoveNext()`



I have already tried:
- running the installation as admin
- unchecking the checkboxes at the start of installation
- enabling the Windows Management Instrumentation service as admin
- removing the docker folders on C: manually

**Comment by awsactivators** [comment]
Hello Docker, I'm trying to install docker desktop 4.11.1 on my windows 11 but it kept failing, I do have the UAC pop up, I have tried via winget, but still the same error. please check the error message that I got below

/Installation failed
Component CommunityInstaller.EnableFeaturesAction failed: Invalid namespace
                                                                               at CommunityInstaller.InstallWorkflow.<DoHandleD4WPackageAsync>d__30.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at CommunityInstaller.InstallWorkflow.<DoProcessAsync>d__23.MoveNext()
Installer failed with exit code: 1

**Comment by icecreamwala12** [comment]
getting the same issue as awsactivators

**Comment by icecreamwala12** [comment]
installation of 4.11.1 was giving issue. so i downloaded the older docker. some version of 3.
it worked for me.

**Comment by awsactivators** [comment]
Will try this out thanks.


> installation of 4.11.1 was giving issue. so i downloaded the older docker. some version of 3.
> 
> it worked for me. 
> 
> then showed me update. and that update finally installed 4.11.1 for me.
> 
> 
> 
> not ideal but it is what it is.

**Comment by icecreamwala12** [comment]
Have you done installation of wsl 2

**Comment by awsactivators** [comment]
Yes I have 

> Have you done installation of wsl 2

**Comment by devang-s** [comment]
I am getting this error installing Docker 4.12.0
```

Component CommunityInstaller.EnableFeaturesAction failed: Not found 
   at CommunityInstaller.InstallWorkflow.<DoHandleD4WPackageAsync>d__30.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at CommunityInstaller.InstallWorkflow.<DoProcessAsync>d__23.MoveNext()
```

**Comment by alaomichael** [comment]
I am getting this error installing Docker 4.14.0

Component CommunityInstaller.EnableFeaturesAction failed: Not found 
   at CommunityInstaller.InstallWorkflow.<DoHandleD4WPackageAsync>d__30.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at CommunityInstaller.InstallWorkflow.<DoProcessAsync>d__23.MoveNext()

**Comment by ddimasv** [comment]
I am getting this error installing Docker 4.14.0

Component CommunityInstaller.EnableFeaturesAction failed: 
   at CommunityInstaller.InstallWorkflow.<DoHandleD4WPackageAsync>d__30.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at CommunityInstaller.InstallWorkflow.<DoProcessAsync>d__23.MoveNext()

**Comment by mat007** [comment]
I am going to close this issue because the OP has found a workaround long ago and the last error messages are very different from the original one, which was
```
Component CommunityInstaller.ExecAction failed: The requested operation requires elevation
```

Please open a separate issue if needed.
And remember: if the error message is different, it’s a different problem.

**Comment by darq13** [comment]
I had the same problem and I went through all these steps and it kept failing. This is what helped for me: 
Type these commands in cmd
1. sc config winmgmt start=disabled
2. net stop winmgmt
3. Winmgmt /salvagerepository %windir%\System32\wbem
4. Winmgmt /resetrepository %windir%\System32\wbem
5. sc config winmgmt start=auto

For more details check this [stackoverflow thread](https://stackoverflow.com/questions/71095210/installing-docker-desktop-4-5-0-failed-componet-communityinstaller-enablefeatur). I just wanted to leave this solution here as well because I struggled so much with this installation.
